kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -518,7 +518,7 @@ public String getFieldQueryParam() {
 
   public L list() throws KubernetesClientException {
     try {
-      HttpUrl.Builder requestUrlBuilder = HttpUrl.get(getNamespacedUrl()).newBuilder();
+      HttpUrl.Builder requestUrlBuilder = HttpUrl.get(getResourceUrl()).newBuilder();
```

Comment:
Why not? You can always call `.withName("serviceName")`.
For practical use see: https://github.com/hazelcast/hazelcast-kubernetes/blob/master/src/main/java/com/noctarius/hazelcast/kubernetes/ServiceEndpointResolver.java